src/authentication/EmptyCredentialsExtractor.ts

```
@@ -0,0 +1,19 @@
+import type { HttpRequest } from '../server/HttpRequest';
+import type { Credentials } from './Credentials';
+import { CredentialsExtractor } from './CredentialsExtractor';
+
+/**
+ * Extracts the empty credentials, indicating an unauthenticated agent.
+ */
+export class EmptyCredentialsExtractor extends CredentialsExtractor {
+  public async canHandle({ headers }: HttpRequest): Promise<void> {
+    const { authorization } = headers;
+    if (authorization) {
+      throw new Error('Unexpected Authorization scheme.');
```

This should throw an `UnsupportedHttpError` (or a more specific one).